packages/audiodocs/docs/guides/create-your-own-effect.mdx

```diff
@@ -0,0 +1,247 @@
+
+### Android
+
+Case with android is much different, because of the way android is compiled we need to compile our library with whole turbo-module. 
+Firstly, follow [the guide](https://reactnative.dev/docs/the-new-architecture/pure-cxx-modules#android), but replace `CmakeLists.txt` with this content:
+
+```cmake
+cmake_minimum_required(VERSION 3.13)
+
+project(appmodules)
+
+set(ROOT ${CMAKE_SOURCE_DIR}/../../../../..)
+set(AUDIO_API_DIR ${ROOT}/node_modules/react-native-audio-api)
+
+include(${REACT_ANDROID_DIR}/cmake-utils/ReactNative-application.cmake)
+
+target_sources(${CMAKE_PROJECT_NAME} PRIVATE
+  ${ROOT}/shared/NativeAudioProcessingModule.cpp
+  ${ROOT}/shared/MyProcessorNode.cpp
+  ${ROOT}/shared/MyProcessorNodeHostObject.cpp
+)
+
+target_include_directories(${CMAKE_PROJECT_NAME} PUBLIC
+    ${ROOT}/shared
+    ${AUDIO_API_DIR}/common/cpp
+)
+
+add_library(react-native-audio-api SHARED IMPORTED)
+string(TOLOWER ${CMAKE_BUILD_TYPE} BUILD_TYPE_LOWER)
+# we need to find where original library was built
+set_target_properties(react-native-audio-api PROPERTIES IMPORTED_LOCATION
+        ${AUDIO_API_DIR}/android/build/intermediates/merged_native_libs/${BUILD_TYPE_LOWER}/merge${CMAKE_BUILD_TYPE}NativeLibs/out/lib/${CMAKE_ANDROID_ARCH_ABI}/libreact-native-audio-api.so
+)
+target_link_libraries(${CMAKE_PROJECT_NAME} react-native-audio-api android log)
+```
+
+## Final touches
+
+Last part is to finally onboard your custom module to your app, by creating typescript interface that would map c++ layer.
+
+```typescript
+// types.ts
+import { AudioNode, BaseAudioContext } from "react-native-audio-api";
+import { IAudioNode, IBaseAudioContext } from "react-native-audio-api/lib/typescript/interfaces";
+
+export interface IMyProcessorNode extends IAudioNode {
+    gain: number;
+}
+
+export class MyProcessorNode extends AudioNode {
+    constructor(context: BaseAudioContext, node: IMyProcessorNode) {
+        super(context, node);
+    }
+
+    public set gain(value: number) {
+        (this.node as IMyProcessorNode).gain = value;
+    }
+
+    public get gain(): number {
+        return (this.node as IMyProcessorNode).gain;
+    }
+}
+
+declare global {
+    var createCustomProcessorNode: (context: IBaseAudioContext) => IMyProcessorNode;
+}
+```
+
+## Example
+
+```tsx
+import React, { useRef } from 'react';
+import {
+  AudioContext,
+  OscillatorNode,
+} from 'react-native-audio-api';
+import { MyProcessorNode } from './types';
+
+function App() {
+  const audioContextRef = useRef<AudioContext | null>(null);
+  if (!audioContextRef.current) {
+    audioContextRef.current = new AudioContext();
+  }
+  const oscillator = audioContextRef.current.createOscillator();
+  // constructor is put in global scope
+  const processor = new MyProcessorNode(audioContextRef.current, global.createCustomProcessorNode(audioContextRef.current.context));
```

Comment:
hmm, would it make sense to wrap the call to global.create in a function exported somewhere? :) 

```ts
import { createCustomProcessorNode } from './creator';
const processor = createCustomProcessorNode(audioContextRef.current);
```

Reply:
Also you're using most of the nodes through useRef, while the processor is used directly, either do not use the refs at all or use it for the processor too. (The first option is preferred)